### PR TITLE
Crossovers

### PR DESCRIPTION
**Pull request by sakex**

### CODE REVIEW

**Comment by sakex** [comment]
First MR with experimental crossovers

File: src/topology/topology.rs [hunk 1]
```diff
@@ -697,6 +698,51 @@ where
...
+                final_gene.memory_weight =
+                    (worst_gene.memory_weight + final_gene.memory_weight) / 2.0;
+                final_gene.reset_input_weight =
+                    (worst_gene.reset_input_weight + final_gene.reset_input_weight) / 2.0;
+                final_gene.update_input_weight =
+                    (worst_gene.update_input_weight + final_gene.update_input_weight) / 2.0;
+                final_gene.reset_memory_weight =
+                    (worst_gene.reset_memory_weight + final_gene.reset_memory_weight) / 2.0;
+                final_gene.update_memory_weight =
+                    (worst_gene.update_memory_weight + final_gene.update_memory_weight) / 2.0;
```
Comment (Nereuxofficial) [inline src/topology/topology.rs:722]: I'd probably write this in a seperate function. Maybe average_genes(&self, other: &gene)

**Review by Nereuxofficial — APPROVED** [review]
Had a minor comment, which would make the code a bit cleaner but other than that: LGTM

**Review by Nereuxofficial — APPROVED** [review]

File: src/topology/topology.rs (hunk 1 above)
Comment (sakex) [reply src/topology/topology.rs]: Thanks it's fixed, I'm going to test it a bit further before publishing to crates.io

**Review by Nereuxofficial — APPROVED** [review]
The version name change makes sense as there's are not any major bugs i can find and feature-wise i can't really tell what's missing

**Pull request merged**